docs/examples/spectrum/spectral_factor.py

```diff
@@ -0,0 +1,143 @@
+# models in pvlib-python to calculate the spectral mismatch factor, :math:`M`.
+# While :math:`M` for a photovoltaic (PV) module can be calculated exactly
+# using spectral irradiance and module spectral response data, these data are
+# often not available. pvlib-python provides several functions to estimate the
+# spectral mismatch factor, :math:`M`, using proxies of the prevailing spectral
+# irradiance conditions, such as air mass and clearsky index, which are easily
+# derived from common ground-based measurements such as broadband irradiance.
+# More information on a range of spectral factor models, as well as the
+# variables upon which they are based, can be found in [1]_.
+#
+# Let's import some data. This example uses a Typical Meteorological Year 3
+# (TMY3) file for the location of Greensboro, North Carolina, from the
+# pvlib-python data directory. This TMY3 file is constructed using the median
+# month from each year between 1980 and 2003, from which we extract the first
+# week of August 2001 to analyse.
+
+# %%
+import pathlib
+from matplotlib import pyplot as plt
+import pandas as pd
+import pvlib
+from pvlib import location
+
+DATA_DIR = pathlib.Path(pvlib.__file__).parent / 'data'
+meteo, metadata = pvlib.iotools.read_tmy3(DATA_DIR / '723170TYA.CSV',
+                                          coerce_year=2001, map_variables=True)
+meteo = meteo.loc['2001-08-01':'2001-08-07']
+
+# %%
+# Spectral Factor Functions
+# -------------------------
+# This example demonstrates the application of three pvlib-python
+# spectral factor functions:
+#
+# - :py:func:`~pvlib.spectrum.spectral_factor_sapm`, which requires only
+#   the absolute airmass, :math:`AM_a`
+# - :py:func:`~pvlib.spectrum.spectral_factor_pvspec`, which requires
+#   :math:`AM_a` and the clearsky index, :math:`k_c`
+# - :py:func:`~pvlib.spectrum.spectral_factor_firstsolar`, which requires
+#   :math:`AM_a` and the atmospheric precipitable water content, :math:`W`
+
+# %%
+# Calculation of inputs
+# ---------------------
+# Let's calculate the absolute air mass, which is required for all three
+# models. We use the Kasten and Young [2]_ model, which requires the apparent
+# sun zenith. Note: TMY3 files timestamps indicate the end of the hour, so we
+# shift the indices back 30-minutes to calculate solar position at the centre
+# of the interval.
+
+# Create a location object
+lat, lon = metadata['latitude'], metadata['longitude']
+alt = altitude = metadata['altitude']
+tz = 'Etc/GMT+5'
+loc = location.Location(lat, lon, tz=tz, name='Greensboro, NC')
+
+# Calculate solar position parameters
+solpos = loc.get_solarposition(
+    meteo.index.shift(freq="-30min"),
+    pressure=meteo.pressure*100,  # convert from millibar to Pa
+    temperature=meteo.temp_air)
+solpos.index = meteo.index  # reset index to end of the hour
+
+amr = pvlib.atmosphere.get_relative_airmass(solpos.apparent_zenith).dropna()
+# relative airmass
+ama = amr*(meteo.pressure/1013.25)  # absolute airmass
```

Comment:
@AdamRJensen referring to your [`ama` comment](https://github.com/pvlib/pvlib-python/pull/2114/files#r1666703553), it starts here. Do you suggest I change amr as well, or just ama?